### PR TITLE
Add support for conflict_target: {:unsafe_fragment, _}

### PR DESCRIPTION
**Pull request by ewildgoose**
Sqlite does support ON CONFLICT DO upsert feature, eg:
  https://www.sqlite.org/lang_UPSERT.html

As part of this I synced a couple of function definitions to match current upstream postgrex as part of this. This is a bit stark as they use a different formatting option. Does this fit with your goals? 

Note my goal here was to be able to do something like:

    Repo.insert(
      %__MODULE__{
        device_id: device_id,
        sim_id: sim_id,
        device_type: device_type
      },
      on_conflict: {:replace_all_except, [:id]},
      conflict_target: {:unsafe_fragment, ~s<("device_id",coalesce("sim_id", ""))>},
      returning: true

Where the goal is to simulate a unique constraint on a column which accepts nulls (default for sqlite is to allow multiple "unique" rows in a nullable column)

### CODE REVIEW

File: lib/ecto/adapters/sqlite3/connection.ex [hunk 1]
```diff
@@ -873,20 +846,11 @@ defmodule Ecto.Adapters.SQLite3.Connection do
...
-        do: update_op(op, update_key(type, key, query, sources), value, sources, query)
-      )
-
-    Enum.intersperse(fields, ", ")
-  end
-
-  defp update_key(_kind, key, _query, _sources) do
-    quote_name(key)
+  defp update_fields(%{updates: updates} = query, sources) do
+    for(%{expr: expr} <- updates,
```
Comment (warmwaffles) [inline lib/ecto/adapters/sqlite3/connection.ex:850]: Did `mix format` make this get all aligned on the same column?

**Review by warmwaffles — APPROVED** [review]
This looks good to me. Are there tests we could add to the integration suite to ensure this is doing what we want it to do? Or rather are there already existing integration tests that we can enable to test against?

**Comment by warmwaffles** [comment]
> Sorry, I keep saying "postgrex", when I really mean ecto_sql/postgres

Don't worry it confuses me as well.

**Comment by ewildgoose** [comment]
I presume it's this change causing the issue (or the update_fields() change, but can't see how that causes it?)

```
-    [
-      cte,
-      prefix,
-      fields,
-      join,
-      where,
-      returning(query, sources)
-    ]
+    [cte, prefix, fields, join, where | returning(query, sources)]
```

**Pull request closed without merging**